vectortile/src/main/scala/geotrellis/vectortile/protobuf/ProtobufTile.scala

```diff
@@ -0,0 +1,293 @@
+    }
+
+    loop(feats)
+  }
+
+  /**
+   * Construct Feature-specific metadata from the key/value lists of
+   * the parent layer.
+   */
+  private def getMeta(keys: Seq[String], vals: Seq[vt.Tile.Value], tags: Seq[Int]): Map[String, Value] = {
+    val pairs = new ListBuffer[(String, Value)]
+    var i = 0
+
+    while (i < tags.length) {
+      val k: String = keys(tags(i))
+      val v: vt.Tile.Value = vals(tags(i + 1))
+
+      pairs.append(k -> v)
+
+      i += 2
+    }
+
+    pairs.toMap
+  }
+
+  /* Geometry Streams */
+  private lazy val pointStream = geomStream[Point, MultiPoint](pointFs)
+  private lazy val lineStream = geomStream[Line, MultiLine](lineFs)
+  private lazy val polyStream = geomStream[Polygon, MultiPolygon](polyFs)
+
+  /* OPTIMIZATION NOTES
+   * `Stream.flatMap` maintains laziness. A common pattern here to "fold away"
+   * results you don't want are to use [[Option]]. However, flatMap here
+   * expects an [[Iterable]], and employs an implicit conversion from [[Option]]
+   * to get it.
+   *
+   * By calling directly what that implicit eventually calls at the bottom of
+   * its call stack, we save some operations.
+   */
+  lazy val points: Stream[Feature[Point, Map[String, Value]]] = pointStream
+    .flatMap({
+      case (Left(p), meta) => new ::(Feature(p, meta), Nil)
+      case _ => Nil
+    })
+
+  lazy val multiPoints: Stream[Feature[MultiPoint, Map[String, Value]]] = pointStream
+    .flatMap({
+      case (Right(p), meta) => new ::(Feature(p, meta), Nil)
+      case _ => Nil
+    })
+
+  lazy val lines: Stream[Feature[Line, Map[String, Value]]] = lineStream
+    .flatMap({
+      case (Left(p), meta) => new ::(Feature(p, meta), Nil)
+      case _ => Nil
+    })
+
+  lazy val multiLines: Stream[Feature[MultiLine, Map[String, Value]]] = lineStream
+    .flatMap({
+      case (Right(p), meta) => new ::(Feature(p, meta), Nil)
+      case _ => Nil
+    })
+
+  lazy val polygons: Stream[Feature[Polygon, Map[String, Value]]] = polyStream
+    .flatMap({
+      case (Left(p), meta) => new ::(Feature(p, meta), Nil)
+      case _ => Nil
+    })
+
+  lazy val multiPolygons: Stream[Feature[MultiPolygon, Map[String, Value]]] = polyStream
+    .flatMap({
+      case (Right(p), meta) => new ::(Feature(p, meta), Nil)
+      case _ => Nil
+    })
+
+  /**
+   * Given a raw protobuf Layer, segregate its Features by their GeomType.
+   * `UNKNOWN` geometry types are ignored.
+   */
+  private def segregate(
+    features: Seq[vt.Tile.Feature]
+  ): (ListBuffer[vt.Tile.Feature], ListBuffer[vt.Tile.Feature], ListBuffer[vt.Tile.Feature]) = {
+    val points = new ListBuffer[vt.Tile.Feature]
+    val lines = new ListBuffer[vt.Tile.Feature]
+    val polys = new ListBuffer[vt.Tile.Feature]
+
+    features.foreach { f =>
+      f.getType match {
+        case POINT => points.append(f)
+        case LINESTRING => lines.append(f)
+        case POLYGON => polys.append(f)
+        case _ => Unit // `UNKNOWN` or `Unrecognized`.
```

Comment:
Gotcha, I guess that makes sense to just ignore the valid but Unknown values. We should doc that we just skip anything that isn't a geometry or is unknown to us, in case you haven't already done that.